### ISSUE TITLE
What to do with static sites?

### ISSUE THREAD

**Issue by ismay** [bramstein/fontfaceobserver#40]
I understand that it's a common solution to set a cookie and apply a class server side to prevent a FOUT when fonts are already loaded. But how would you solve this for static sites?

Right now fontface observer is working very well, but because the html isn't processed server side, I've no way to add the `fontsLoaded` class before it reaches the client. So currently I'm seeing a FOUT for every page load (for example this [site](https://ismaywolff.nl) and the matching [repo](https://github.com/ismay/ismaywolff.nl))

**Comment by robwierzbowski** [comment]
I set a `sessionStorage` key to true on font load, then check it in an inline script in the head of the document. If it's true, I set a class on the document root (`html`) element. Works pretty well for single sessions (there's a FOUT every new session even if the font is in the cache, although it's really short), but there's no true connection between whether the font is in the cache and the `sessionStorage` value.

If you want to optimize for return visitors it seems like the best option is to use `localStorage` and re-implement caching by saving the font as base64 in CSS, and saving the CSS in `localStorage`.
